TrackingTools/TrajectoryFiltering/interface/SeedExtensionTrajectoryFilter.h

```diff
@@ -0,0 +1,51 @@
+#ifndef SeedExtensionTrajectoryFilter_H
+#define SeedExtensionTrajectoryFilter_H
+
+#include "TrackingTools/TrajectoryFiltering/interface/TrajectoryFilter.h"
+
+class SeedExtensionTrajectoryFilter final : public TrajectoryFilter {
+public:
+
+  explicit SeedExtensionTrajectoryFilter() {} 
+  
+  explicit SeedExtensionTrajectoryFilter(edm::ParameterSet const & pset, edm::ConsumesCollector&) :
+     theStrict(pset.existsAs<bool>("strictSeedExtension") ? pset.getParameter<bool>("strictSeedExtension"):false),
```

Comment:
So let me repeat the story (again, and maybe some day we get there). The inherent "problem" is that the tracking code is organized as EDProducers that make extensive use of helper plugins (that use plugins that use plugins; by very quick look I can indeed find three layers of these...), and I'm not aware of any "easy" solution for fillDescription for plugins. E.g. the `TrajectoryFilter`s here are such plugins.

One option suggested by @Dr15Jones 
https://hypernews.cern.ch/HyperNews/CMS/get/edmFramework/3446/1.html
would be to create a second class structures only for the fillDescriptions (my impression of the thread above is that the validation system does not currently understand `refToPSet_`). Back when we decided to replace the ESProducers with helper plugins and `refToPSet_` we had many other alternatives on the table
https://indico.cern.ch/event/308199/contribution/8/2/attachments/588340/809738/slides.pdf
https://hypernews.cern.ch/HyperNews/CMS/get/edmFramework/3244/4/1.html
I'm repeating these because one of the options back then was to store the helper algorithms to event (EDProducers could naturally use fillDescriptions), but downsides being either
* no finer-grained caching than event (e.g. seed), or
* rewriting algorithms to separate the transient state and pass that around
* (or we could try to see if all fine-grained caching is still needed, I don't anymore remember which pieces of code do it)

Clearly whatever we will for the fillDescriptions in tracking will require significant development investment (possibly having impact on HLT configuration),  and then somebody need(s) to decide what to do and when to do it.

As for HLT and `existsAs`, why not employing `customizeHLTforCMSSW.py`?